packages/go_router/test/logging_test.dart

```diff
@@ -0,0 +1,29 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter_test/flutter_test.dart';
+import 'package:go_router/src/logging.dart';
+import 'package:logging/logging.dart';
+
+void main() {
```

Comment:
We could test the behavior where specifying `debugLogDiagnostics` calls setLogging with the correct value. We could also validate the log output itself, but that might be overkill.

Reply:
If logging is disabled, would there be 0 logs? If yes, we could test for the absence of logs instead of validating log messages directly. If no, feel free to skip as checking whether `setLogging` was called with the correct value is perfectly reasonable.